Anonymize/Type/ObjectAnonymizer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * This file is part of the GDPR bundle.
+ *
+ * @category  Bundle
+ * @package   Gdpr
+ * @author    SuperBrave <info@superbrave.nl>
+ * @copyright 2018 SuperBrave <info@superbrave.nl>
+ * @license   https://github.com/superbrave/gdpr-bundle/blob/master/LICENSE MIT
+ * @link      https://www.superbrave.nl/
+ */
+
+namespace SuperBrave\GdprBundle\Anonymize\Type;
+
+use SuperBrave\GdprBundle\Anonymize\Anonymizer;
+
+/**
+ * Class ObjectAnonymizer
+ *
+ * @package SuperBrave\GdprBundle\Anonymize\Type
+ */
+class ObjectAnonymizer implements AnonymizerInterface
+{
+    /**
+     * @var Anonymizer
+     */
+    private $anonymizer;
+
+    /**
+     * ObjectAnonymizer constructor.
+     *
+     * @param Anonymizer $anonymizer Any Anonymizer
+     */
+    public function __construct(Anonymizer $anonymizer)
+    {
+        $this->anonymizer = $anonymizer;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException if propertyValue is not an object.
+     */
+    public function anonymize($propertyValue, array $options = [])
+    {
+        if (!is_object($propertyValue)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Invalid argument given; "%s" for class "%s" should be of type object.',
+                gettype($propertyValue),
+                __CLASS__
+            ));
+        }
+
+        $this->anonymizer->anonymize($propertyValue);
```

Comment:
The return value of anonymize() should be assigned here.

Reply:
$propertyValue is an object in this class so it changes is by reference!
Hence why it returns the object without directly assigning it.

Reply:
I smell an inconsistency, because ArrayAnonymizer does assign the return values from its anonymizer. If both are by design, why the difference?

Reply:
ArrayAnonymizer is a wrapper around another anonymizerInterface implementation. You can have an array of datetimes, or an array of ip addresses, or an array of objects. By injecting one anonymizer implementation into the ArrayAnonymizer you can build the anonymizer you need. This is already done for the CollectionAnonymizer, what an ArrayAnonymizer with the ObjectAnonymizer is. The ObjectAnonymizer in turn does not use another anonymizerInterface immplementation, but uses the Anonymizer class to (recursively) anonymize objects.